language/stdlib/modules/multi_account.move

```diff
@@ -0,0 +1,104 @@
+// Errors:
+// 0 -> ADDRESS_NOT_ALLOWED
+// 1 -> CURRENCY_NOT_HELD
+address 0x0:
+module MultiCurrencyHolding {
+    use 0x0::LibraAccount;
+    use 0x0::Transaction;
+    use 0x0::Libra;
+    use 0x0::Vector;
+
+    // A resource that specifies the addresses that are allowed to be added
+    // to this multi-currency holding.
+    resource struct ChildInfo {
+        allowed_addresses: vector<address>,
+    }
+
+    // Withdrawal capabilities for the added accounts, partitioned by their
+    // currency type.
+    resource struct AccountInfo<Token> {
+        withdrawal_caps: vector<LibraAccount::WithdrawalCapability>,
+    }
+
+    // Allow the account at `addr` to delegate its
+    // withdrawal_capability to us.
+    public fun allow_account_address(addr: address)
+    acquires ChildInfo {
+        let sender = Transaction::sender();
+        if (!exists<ChildInfo>(sender)) move_to_sender(ChildInfo{
+            allowed_addresses: Vector::empty()
+        });
+        let child_info = borrow_global_mut<ChildInfo>(sender);
+        if (!Vector::contains(&child_info.allowed_addresses, &addr))
+            Vector::push_back(&mut child_info.allowed_addresses, addr);
+    }
+
+    // Allow holding currencies of `Token` type in this multi-currency
+    // holding.
+    public fun allow_currency<Token>() {
+        move_to_sender(AccountInfo<Token>{ withdrawal_caps: Vector::empty() })
+    }
+
+    // Add the sending account of currency type `Token` to the
+    // multi-currency holding with parent address at `parent_account_addr`.
+    // The sending address must be in the `allowed_addresses` held under
+    // the parent account.
+    public fun add_account_to<Token>(parent_account_addr: address)
+    acquires ChildInfo, AccountInfo {
+        let sender = Transaction::sender();
+        let child_info = borrow_global_mut<ChildInfo>(parent_account_addr);
+        Transaction::assert(Vector::contains(&child_info.allowed_addresses, &sender), 0);
+        let account_info = borrow_global_mut<AccountInfo<Token>>(parent_account_addr);
+        Vector::push_back(
+            &mut account_info.withdrawal_caps,
+            LibraAccount::extract_sender_withdrawal_capability()
+        );
```

Comment:
Should this also remove `sender`  from `allowed_addresses`? This makes it easier to see which children are added vs pending, and will be better from a space perspective.